src/V3Param.cpp

```diff
@@ -769,57 +785,61 @@ class ParamProcessor final {
         if (debug() >= 10) nodep->dumpTree(cout, "-cell: ");
         // Evaluate all module constants
         V3Const::constifyParamsEdit(nodep);
-        AstNodeModule* const srcModp = nodep->modp();
         srcModp->someInstanceName(someInstanceName + "." + nodep->name());
 
         // Make sure constification worked
         // Must be a separate loop, as constant conversion may have changed some pointers.
         // if (debug()) nodep->dumpTree(cout, "-cel2: ");
         string longname = srcModp->name() + "_";
         bool any_overrides = false;
-        // Must always clone __Vrcm (recursive modules)
-        if (nodep->recursive()) any_overrides = true;
-        if (debug() > 8 && nodep->paramsp())
-            nodep->paramsp()->dumpTreeAndNext(cout, "-cellparams: ");
+        AstPin* paramsp = nullptr;
+        if (const auto* cellp = VN_CAST(nodep, Cell)) {
```

Comment:
Wouldn't it be more readable to have separate a `deparam` function for each case and then the common bits in other functions we call into? That would reduce the ifology a bit.

Reply:
It is a bit complicated.  We need a bunch of ifs anyway in order to call the correct function in the first place, and then every function would need to do take 6+ arguments, as the generic logic is quite tightly interleaved with the specialized logic.  I will push an example idea to class-param-rework when I am finished with addressing these comments.